tests/nvapi_sysinfo.cpp

```diff
@@ -813,8 +876,74 @@ TEST_CASE("Sysinfo methods succeed", "[.sysinfo]") {
+                .LR_SIDE_EFFECT(*_3 = temp + 1)
+                .RETURN(NVML_SUCCESS);
+
+            SetupResourceFactory(std::move(dxgiFactory), std::move(vulkan), std::move(nvml), std::move(lfx));
+            REQUIRE(NvAPI_Initialize() == NVAPI_OK);
+
+            NvPhysicalGpuHandle handle;
+            REQUIRE(NvAPI_SYS_GetPhysicalGpuFromDisplayId(0, &handle) == NVAPI_OK);
+
+            SECTION("GetThermalSettings (V1) returns OK") {
+                NV_GPU_THERMAL_SETTINGS_V1 settings;
+                settings.version = NV_GPU_THERMAL_SETTINGS_VER_1;
+                REQUIRE(NvAPI_GPU_GetThermalSettings(handle, NVAPI_THERMAL_TARGET_ALL, reinterpret_cast<NV_GPU_THERMAL_SETTINGS*>(&settings)) == NVAPI_OK);
+                REQUIRE(settings.count == 1);
+                REQUIRE(settings.sensor[0].controller == NVAPI_THERMAL_CONTROLLER_GPU_INTERNAL);
+                REQUIRE(settings.sensor[0].target == NVAPI_THERMAL_TARGET_GPU);
+                REQUIRE(settings.sensor[0].currentTemp == temp);
+                REQUIRE(settings.sensor[0].defaultMaxTemp == maxTemp);
+                REQUIRE(settings.sensor[0].defaultMinTemp == 0U);
+            }
+
+            SECTION("GetThermalSettings (V2) returns OK") {
+                NV_GPU_THERMAL_SETTINGS_V2 settings;
+                settings.version = NV_GPU_THERMAL_SETTINGS_VER_2;
+                REQUIRE(NvAPI_GPU_GetThermalSettings(handle, NVAPI_THERMAL_TARGET_ALL, &settings) == NVAPI_OK);
+                REQUIRE(settings.count == 1);
+                REQUIRE(settings.sensor[0].controller == NVAPI_THERMAL_CONTROLLER_GPU_INTERNAL);
+                REQUIRE(settings.sensor[0].target == NVAPI_THERMAL_TARGET_GPU);
+                REQUIRE(settings.sensor[0].currentTemp == static_cast<int>(temp));
+                REQUIRE(settings.sensor[0].defaultMaxTemp == static_cast<int>(maxTemp));
+                REQUIRE(settings.sensor[0].defaultMinTemp == -40);
+            }
+
+            SECTION("GetThermalSettings with unknown struct version returns incompatible-struct-version") {
+                NV_GPU_THERMAL_SETTINGS settings;
+                settings.version = NV_GPU_THERMAL_SETTINGS_VER_2 + 1;
+                REQUIRE(NvAPI_GPU_GetThermalSettings(handle, NVAPI_THERMAL_TARGET_ALL, &settings) == NVAPI_INCOMPATIBLE_STRUCT_VERSION);
+            }
+
+            SECTION("GetThermalSettings with current struct version returns not incompatible-struct-version") {
+                // This test should fail when a header update provides a newer not yet implemented struct version
+                NV_GPU_THERMAL_SETTINGS settings;
+                settings.version = NV_GPU_THERMAL_SETTINGS_VER;
+                REQUIRE(NvAPI_GPU_GetThermalSettings(handle, NVAPI_THERMAL_TARGET_ALL, &settings) != NVAPI_INCOMPATIBLE_STRUCT_VERSION);
+            }
+        }
+
+        SECTION("GetThermalSettings succeeds when DeviceGetTemperature is available") {
```

Comment:
Since this is the fallback, naming it like `GetThermalSettings succeeds when DeviceGetThermalSettings is not available` fits better.

Reply:
I'd actually prefer to be more verbose here and have something like `GetThermalSettings succeeds when DeviceGetThermalSettings is not available but DeviceGetTemperature is` so we don't claim that it's meant to succeed when local version of NVML supports neither function.

Reply:
Yeah that's even better.